### ISSUE TITLE
Variables are not correctly resolved in Postman VS Code App

### ISSUE THREAD

**Issue by rknabben** [a85/POSTMan-Chrome-Extension#12179]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

I have an environment setup with a variable called BaseURL containing https://marketplace.also.nl/SimpleAPI/SimpleAPIService.svc/rest/

I am calling the API's with {{BaseURL}}/GetSessionToken
In Postman for Windows, no problem. The API call is performed and all is right.

![image](https://github.com/postmanlabs/postman-app-support/assets/40603570/ea982f9f-443b-478a-9dc3-ba0cf9fa50d5)

In the Postman Extension for Visual Studio Code there is an error: getaddrinfo ENOTFOUND {{baseurl}}
Hovering above the Variable only reveals: "https://marketplace.also." 
the Curl code is correct (full URL), but as said the request is not completed

![image](https://github.com/postmanlabs/postman-app-support/assets/40603570/2727615b-0a67-4e89-96af-97d06902ddae)

### Steps To Reproduce

1. Create an environment containing the variable {{BaseURL}}
2. Create a new request with an URL containing the {{BaseURL}} variable.
3. Try to execute the request

### Operating System

Windows

### Postman Version

Windows : 10.16 (july 2023)   VSCODE extension: 0.4.1 (on VSCODE 1.80.1)

### Postman Platform

Other

### User Account Type

Signed In User

**Comment by coditva** [comment]
@rknabben Thanks for reaching out. Can you help me with a few more things that would help me in replicating this issue:
1. Can you set this variable in `Globals` or `Collection` and see if this is resolving?
2. Are you consistently able to replicate or was this a one-off issue.
3. Do you see any logs or failed network calls in VS Code Developer Tools (Can be found in `Help` > `Toggle Developer Tools`).
4. Is the error that you're getting `getaddrinfo ENOTFOUND {{baseurl}}` or `getaddrinfo ENOTFOUND {{BaseURL}}`? There could be some case specific error here.

**Comment by rknabben** [comment]
Sure I can ->

  1.  Can you set this variable in Globals or Collection and see if this is resolving?
Collection -> Resolves in the URL.
The BODY variables “username” : “{{username}}” never resolves. Not with Collection, not with Environment.
I have no idea how to set a global variable from within the VSCODE App.

  1.  Are you consistently able to replicate or was this a one-off issue.
No, it doesn’t work for any of my PostMan collections. (All depend on the Environments)

  1.  Do you see any logs or failed network calls in VS Code Developer Tools (Can be found in Help > Toggle Developer Tools).
Lots of:
DevTools failed to load source map: Could not load content for https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/XXXXX/.vscode/extensions/postman.postman-for-vscode-0.4.1/dist/http-workbench/5634.9510e52a844fa32a.js.map: Connection error: net::ERR_NAME_NOT_RESOLVED
DevTools failed to load source map: Could not load content for https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/XXXXX/.vscode/extensions/postman.postman-for-vscode-0.4.1/dist/context-bar/remoteEntry.js.map: Connection error: net::ERR_NAME_NOT_RESOLVED
DevTools failed to load source map: Could not load content for https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/XXXXX/.vscode/extensions/postman.postman-for-vscode-0.4.1/dist/context-bar/7769.db6db35e1517dc15.js.map: Connection error: net::ERR_NAME_NOT_RESOLVED
DevTools failed to load source map: Could not load content for https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/XXXXX/.vscode/extensions/postman.postman-for-vscode-0.4.1/dist/context-bar/5650.f2866f29ef08b365.js.map: Connection error: net::ERR_NAME_NOT_RESOLVED


  1.  Is the error that you're getting getaddrinfo ENOTFOUND {{baseurl}} or getaddrinfo ENOTFOUND {{BaseURL}}? There could be some case specific error here.
Good point ->  getaddrinfo ENOTFOUND {{baseurl}}  so all lowercase. (Changing the variablename to all lowercase still does not work)



Van: Utkarsh Maheshwari ***@***.***>
Verzonden: maandag 24 juli 2023 07:47
Aan: postmanlabs/postman-app-support ***@***.***>
CC: Rob Knabben | R2-4U ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [postmanlabs/postman-app-support] Variables are not correctly resolved in Postman VS Code App (Issue #12179)


@rknabben<https://github.com/rknabben> Thanks for reaching out. Can you help me with a few more things that would help me in replicating this issue:

  1.  Can you set this variable in Globals or Collection and see if this is resolving?
  2.  Are you consistently able to replicate or was this a one-off issue.
  3.  Do you see any logs or failed network calls in VS Code Developer Tools (Can be found in Help > Toggle Developer Tools).
  4.  Is the error that you're getting getaddrinfo ENOTFOUND {{baseurl}} or getaddrinfo ENOTFOUND {{BaseURL}}? There could be some case specific error here.

**Comment by coditva** [comment]
@rknabben Although we've not been able to replicate the exact issue that you're facing, we've updated the extension to include some fixes in this area. Can you update (to `0.5.1`) and check if the issue still persists? If it still persists, we'll also be interested to get on a call and debug the issue to resolve this faster.

<img width="541" alt="image" src="https://github.com/postmanlabs/postman-app-support/assets/8445768/02c435f8-bfd5-4f61-b447-35b1d09011e2">

Thanks.

**Comment by rknabben** [comment]
I’ve updated to 0.5.1 and the problem seems to be PARTLY fixed, so that’s a good start :


In the environment:

***@***.***

In the URL / Body:

***@***.***

URL When typing: (correctly resolved)

***@***.***

URL when hovering: (only partly resolved, but it might also be a word wrapping issue? Overflow = hidden instead of overflow = ellipsis )
***@***.***


When Executing the query in Postman Windows it works (left/white), and returns an Token, when the exact same query is performed in the VSCode APP (right/dark) , it apparently does not send the right username + password.
But…as you can see, the URL is now resolved and the correct API is called, so that part is resolved 😊

***@***.***

So Conclusion:

The URL Resolver now works, but the body variable resolving does not. Or at least seems to do not 😊


Van: Utkarsh Maheshwari ***@***.***>
Verzonden: woensdag 26 juli 2023 14:35
Aan: postmanlabs/postman-app-support ***@***.***>
CC: Rob Knabben | R2-4U ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [postmanlabs/postman-app-support] Variables are not correctly resolved in Postman VS Code App (Issue #12179)


@rknabben<https://github.com/rknabben> Although we've not been able to replicate the exact issue that you're facing, we've updated the extension to include some fixes in this area. Can you update (to 0.5.1) and check if the issue still persists? If it still persists, we'll also be interested to get on a call and debug the issue to resolve this faster.
[image]<https://user-images.githubusercontent.com/8445768/256231567-02c435f8-bfd5-4f61-b447-35b1d09011e2.png>

Thanks.

**Comment by coditva** [comment]
@rknabben Unfortunately, the reply that you sent has the values obfuscated (I'm guessing GitHub is masking all the emails that it finds in the reply). Can you update the comment?

Also, can you check what the "Initial" and "Current" values are in the environment and if only the "Current" value is not resolving or any value is not resolving?

In case you're available, we can get on a call as well to debug this.

**Comment by rknabben** [comment]
I’ve updated to 0.5.1 and the problem seems to be PARTLY fixed, so that’s a good start : 


In the environment:

![screenshot1](https://github.com/postmanlabs/postman-app-support/assets/40603570/2f18b1c6-f5d2-4a2f-a24c-d016e7cd286c)

In the URL / Body:

![screenhot2](https://github.com/postmanlabs/postman-app-support/assets/40603570/30a33226-1454-4c7e-bb31-2b0ba98f0728)

URL When typing: (correctly resolved)

![screenshot3](https://github.com/postmanlabs/postman-app-support/assets/40603570/d2be0372-1252-47c1-90f5-6b0461c44b10)

URL when hovering: (only partly resolved, but it might also be a word wrapping issue? Overflow = hidden instead of overflow = ellipsis )

![image](https://github.com/postmanlabs/postman-app-support/assets/40603570/c4e13c6d-f5e0-424a-b7bd-3087639a655e)
 
When Executing the query in Postman Windows it works (left/white), and returns an Token, when the exact same query is performed in the VSCode APP (right/dark) , it apparently does not send the right username + password.
But…as you can see, the URL is now resolved and the correct API is called, so that part is resolved 😊

**Comment by rknabben** [comment]
I've created the reply again here, as the email reply refuses to show the images :)
Sorry about the confusion.

**Comment by aminahbl** [comment]
Hey @coditva, in case it's useful feedback: 

I'm using version `0.8.0` and variables work for me in most cases, but I get `Error: getaddrinfo ENOTFOUND {{base_url}}` when running (`Try`) an `example` if I've set the variable in an environment scope.    

The actual request itself (with the exact same `base_url` variable) works fine, it's only when using the `example` that the error pops up. The log gives:

>(2023-09-17T14:02:18.378Z) [warning] [extension:tabs: factory:requestWorkbenchTabController] Received an unexpected message from the webview: {"type":"VIEW_COLLECTION_FOLDER_REQUEST","payload":{"to":"/workspace/…/request/create?example=…&request=…","params":{},"options":{"customNavEvent":"VIEW_COLLECTION_FOLDER_REQUEST"}}}

If I set the variable in a collection, I don't get the error which is perfectly fine for me. :slightly_smiling_face:

**Comment by pankajk07** [comment]
@rknabben Can you please update the extension to the latest version and let us know if the issue is still reproducible?

**Comment by rknabben** [comment]
Hello Pankajk07,

The issue is solved!
The URL is now correctly resolved from the Environment.

![image](https://github.com/user-attachments/assets/f8b57ce8-438c-49c3-bb1f-24c339c84af6)

**Comment by aymarc** [comment]
I faced the same issue even in the current version 1.2 it turns out that it was a **naming convention** issue. Camel case did not work for me, but all lowercase worked.

- this **baseURL** did not work
- But this **baseurl** worked

**Comment by lukebennett** [comment]
Yes, I can confirm that there is still a bug here (I have v1.6.1 of the VS Code extension, running in Cursor) - it appears to be lowercasing the URL before substituting any variables, meaning any variable names containing a capital letter no longer match. I have `{{baseUrl}}` in my URL and a variable with the same name, but the error I get is `Error: getaddrinfo ENOTFOUND {{baseurl}}` (note the lowercase `{{baseurl}}`).

That said, curiously `{{baseurl}}` doesn't work either - but `{{base-url}}` does. Maybe `baseurl` is a reserved term or something?

**Edit:** I've changed this comment a few times as the behaviour changed in follow-up attempts. My theory above clearly isn't right, as I now have it working with `{{baseUrl}}` - I've rotated through several variable names, I can't find a pattern to the behaviour. All I know is that saving the parent collection doesn't appear to have an immediate impact on the new variable name, which doesn't get picked up until some random time later, it's like there's some kind of caching bug but I can't reliably reproduce it.

I don't think this comment will be particularly helpful in diagnosing the issue now, but it does at least indicate to anybody else looking that there's still a problem here.

**Edit 2:** Had the same problem with a collection-level auth header not flowing through to a request. Eventually got it to come through by closing and re-opening the request panel - this seems to work fairly reliably. Does seem as though the bug is nothing to do with variable casing and all to do with some kind of lag in active collection and request panels syncing with each other.
